p2p/src/test/kotlin/maru/p2p/BeaconBlocksByRangeIntegrationTest.kt

```diff
@@ -0,0 +1,285 @@
+/*
+ * Copyright Consensys Software Inc.
+ *
+ * This file is dual-licensed under either the MIT license or Apache License 2.0.
+ * See the LICENSE-MIT and LICENSE-APACHE files in the repository root for details.
+ *
+ * SPDX-License-Identifier: MIT OR Apache-2.0
+ */
+package maru.p2p
+
+import io.libp2p.core.PeerId
+import java.util.concurrent.TimeUnit
+import maru.config.P2P
+import maru.config.consensus.ElFork
+import maru.config.consensus.qbft.QbftConsensusConfig
+import maru.consensus.ConsensusConfig
+import maru.consensus.ForkIdHashProvider
+import maru.consensus.ForkIdHasher
+import maru.consensus.ForkSpec
+import maru.consensus.ForksSchedule
+import maru.core.SealedBeaconBlock
+import maru.core.ext.DataGenerators
+import maru.core.ext.metrics.TestMetrics
+import maru.crypto.Hashing
+import maru.database.BeaconChain
+import maru.database.InMemoryBeaconChain
+import maru.p2p.messages.BeaconBlocksByRangeRequest
+import maru.p2p.messages.BeaconBlocksByRangeResponse
+import maru.p2p.messages.StatusMessageFactory
+import maru.serialization.ForkIdSerializers
+import maru.serialization.rlp.RLPSerializers
+import org.apache.tuweni.bytes.Bytes
+import org.assertj.core.api.Assertions.assertThat
+import org.awaitility.Awaitility.await
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.parallel.Execution
+import org.junit.jupiter.api.parallel.ExecutionMode
+import tech.pegasys.teku.networking.p2p.libp2p.LibP2PNodeId
+
+@Execution(ExecutionMode.SAME_THREAD)
+class BeaconBlocksByRangeIntegrationTest {
+  companion object {
+    private val chainId = 1337u
+    private const val IPV4 = "127.0.0.1"
+    private const val PORT1 = 9334u
+    private const val PORT2 = 9335u
```

Comment:
The more we're delaying dynamic port allocation in the tests, the more this approach is copy pasted and  more tech debt we introduce. Maybe we should prioritize [this](https://github.com/Consensys/maru/issues/121) or at least develop new, parallelizable approach for the new tests we introduce?